nautobot/utilities/forms/forms.py

```diff
@@ -324,17 +326,27 @@ def capitalize(field):
 
     def _get_lookup_field_choices(self):
         """Get choices for lookup_fields i.e filterset parameters without a lookup expr"""
+        from nautobot.extras.filters.mixins import RelationshipFilter  # Avoid circular import
+
         filterset_without_lookup = (
-            (name, field.label or self.capitalize(field.field_name))
+            (
+                name,
+                field.label
+                or (isinstance(field, RelationshipFilter) and field.relationship.get_label(side=field.side))
+                or self.capitalize(field.field_name),
+            )
             for name, field in self.filterset_filters.items()
-            if "__" not in name and name != "q"
+            if isinstance(field, RelationshipFilter) or ("__" not in name and name != "q")
         )
         return sorted(filterset_without_lookup, key=lambda x: x[1])
 
 
 def dynamic_formset_factory(filterset_class, data=None, **kwargs):
     filter_form = DynamicFilterForm
-    filter_form.filterset_class = filterset_class
+    if isinstance(filterset_class, type):
```

Comment:
Same here, can we let this function take a `filterset` as an argument instead of permitting an instance as the so-called `filterset_class`?